userland/test/test_publisher.cpp

```diff
@@ -0,0 +1,48 @@
+#include <chrono>
+#include <iostream>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <simple_msgs/Uint32.h>
+
+
+template<typename T>
+int publish(
+  rclcpp::Node::SharedPtr node,
+  void (*set_data_func)(typename T::Ptr&, size_t),
```

Comment:
You might want to use the `std::function` feature of C++11 here instead of the C-style function pointer.